spec/requests/endpoints/token_spec.rb

```diff
@@ -22,10 +22,10 @@
 
   it "accepts client credentials with basic auth header" do
     post token_endpoint_url,
-         params: {
+         params: token_endpoint_params(
```

Comment:
This was failing because of missing `grant_type` parameter. I'm not sure how it ever passed.

Reply:
(it's clearly using `authorization_code` though, due to the presence of the `code` parameter

Reply:
`token_endpoint_url` is a spec helper which has this:

```ruby
 grant_type: options[:grant_type] || "authorization_code",
```

So I believe with `token_endpoint_params` you just override the params from the URL generated by the helper